Add information about array interoperability (#4554)
JetBrains/kotlin-web-site: docs/topics/wasm/wasm-js-interop.md

}
```

### Array interoperability

> You can copy JavaScript's `JsArray<T>` into Kotlin's native `Array` or `List` types and vice versa.

Reviewer: ```suggestion
You can convert JavaScript's `JsArray<T>` to Kotlin's native `Array` or `List` types, and likewise you can convert these Kotlin types to `JsArray<T>`.
```

Reviewer: Mainly to avoid the Latin idiom for non-English speakers and to avoid repeating "the other way around" from the next line.

Also although this way convert is used multiple times, I feel like "copy" could me misunderstood

Author: @skuzmich  Do you think "convert" can work here or would be misunderstood and we prefer "copy"?

Author: I'm leaving as "copy" as per suggested by developer.

> You can directly transform JavaScript's `JsArray<T>` into Kotlin's native `Array` or `List` types and vice versa.

Reviewer 2: I'd replace "transform" to "copy", and drop unnecessary "directly". Would make it a bit more clear that original JsArray is not affected.

> To automatically convert `JsArray<T>` to `Array<T>` or the other way around, use one of the available [adapter functions](https://github.com/Kotlin/kotlinx-browser/blob/dfbdceed314567983c98f1d66e8c2e10d99c5a55/src/wasmJsMain/kotlin/arrayCopy.kt).

Reviewer 2: Nit: remove unnecessary "automatically". For some reason "automatic conversion" reminds me of "implicit type coercion" which Kotlin chooses not to have :)

> Here's an example of conversion between generic types: Kotlin `List<T> `and `Array<T>` to JavaScript `JsArray<T>.`

Reviewer: ```suggestion
Here's an example of a conversion between generic types:
```

Reviewer: I think we could omit the last part here

```kotlin
val list: List<JsString> =
    listOf("Kotlin", "Wasm").map { it.toJsString() }

// Uses .toJsArray() to convert List or Array to JsArray
val jsArray: JsArray<JsString> = list.toJsArray()

// Uses .toArray() and .toList() to convert it back to Kotlin types 
val kotlinArray: Array<JsString> = jsArray.toArray()
val kotlinList: List<JsString> = jsArray.toList()
```

Similar adapter functions are available for converting typed arrays to their Kotlin equivalents
(for example, `IntArray` and `Int32Array`). For detailed information and implementation,
see the [`kotlinx-browser` repository]( https://github.com/Kotlin/kotlinx-browser/blob/dfbdceed314567983c98f1d66e8c2e10d99c5a55/src/wasmJsMain/kotlin/arrayCopy.kt).

> Here's an example of conversion between typed arrays: Kotlin `IntArray` to JavaScript `Int32Array`.

Reviewer: ```suggestion
Here's an example of a conversion between typed arrays:
```

Reviewer: again, I think we can omit the second part here - alternatively, it could be written as:
Here’s an example that demonstrates converting Kotlin's `List<T>` and `Array<T>` to JavaScript JsArray<T>:
